src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -1177,7 +1177,7 @@ internal void EnterScriptFunction(FunctionContext functionContext)
         private void SetupBreakpoints(FunctionContext functionContext)
         {
             var scriptDebugData = _mapScriptToBreakpoints.GetValue(functionContext._sequencePoints,
-                                                                   _ => Tuple.Create(new List<LineBreakpoint>(),
+                                                                   _ => Tuple.Create(new Dictionary<int, List<LineBreakpoint>>(),
```

Comment:
Using non-concurrent dictionary here because the previous approach was also not concurrent.